pkg/sysregistries/system_registries.go

```diff
@@ -0,0 +1,99 @@
+package sysregistries
+
+import (
+	"github.com/ghodss/yaml"
+	"io/ioutil"
+)
+
+/*
+systemRegistriesConfPath is the path to the system-wide registry configuration file
+and is used to add/subtract potential registries for obtaining images
+You can override this at build time with
+-ldflags '-X github.com/containers/image/sysregistries.systemRegistriesConfPath=$your_path'
+*/
+var systemRegistriesConfPath = builtinRegistriesConfPath
+
+/*
+builtinRegistriesConfPath is the path to registry configuration file
+DO NOT change this, instead see systemRegistriesConfPath above.
+*/
+const builtinRegistriesConfPath = "/etc/containers/registries.conf"
+
+var blank_string_array []string
+
+type registries struct {
+	Registries         []string `json:"registries"`
+	InsecureRegistries []string `json:"insecure_registries"`
+	BlockRegistries    []string `json:"block_registries"`
+}
+
+var blankFile = []byte("")
+
+/*
+Reads the global registry file from the filesystem. Returns
+a byte array
+*/
+func readRegistryConf() ([]byte, error) {
+	configBytes, err := ioutil.ReadFile(systemRegistriesConfPath)
+	if err != nil {
+		return blankFile, err
+	}
+	return configBytes, err
+}
+
+var blankRegistries registries
+
+// For mocking in unittests
+var readConf = readRegistryConf
+
+/*
+Loads the registry configuration file from the filesystem and
+then unmarshals it.  Returns the unmarshalled object.
+*/
+func loadRegistryConf() (registries, error) {
+	var config registries
+
+	configBytes, err := readConf()
+	if err != nil {
+		return blankRegistries, nil
+	}
+
+	err = yaml.Unmarshal(configBytes, &config)
+
+	if err != nil {
+		return blankRegistries, err
+	}
+
+	return config, nil
+}
+
+/*
+Returns an array of strings that contain the names
+of the registries as defined in the system-wide
+registries file.  it returns an empty array if none are
+defined
+*/
+func GetRegistries() ([]string, error) {
+	config, err := loadRegistryConf()
+	if err != nil {
+		return blank_string_array, err
+	}
+
+	return config.Registries, nil
+}
+
+/*
+Returns an array of strings that contain the names
+of the insecure registries as defined in the system-wide
+registries file.  it returns an empty array if none are
+defined
+*/
+func GetInsecureRegistries() ([]string, error) {
+	config, err := loadRegistryConf()
+	if err != nil {
+		return blank_string_array, err
+	}
+
+	return config.InsecureRegistries, err
+
+}
```

Comment:
leaving block in the conf file but no function in the Go code. left only for docker.